### PR TITLE
Issue #SB-7025 feat: View all button functionality (Phase I)

### PR DESCRIPTION
**Pull request by ArchanaKumari01**

### CODE REVIEW

File: src/app/client/src/app/app.module.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { SharedModule } from '@sunbird/shared';
 import { Ng2IziToastModule } from 'ng2-izitoast';
 import { PublicModule } from '@sunbird/public';
 import { TelemetryModule } from '@sunbird/telemetry';
+import {SharedFeaturedModule} from '@sunbird/shared-featured';
```
Comment (anoophm) [inline src/app/client/src/app/app.module.ts]: shared-feature

File: src/app/client/src/app/app.module.ts [hunk 2]
```diff
@@ -32,6 +33,7 @@ import { DeviceDetectorModule } from 'ngx-device-detector';
     TelemetryModule.forRoot(),
     DeviceDetectorModule.forRoot(),
     PublicModule,
+    SharedFeaturedModule,
```
Comment (anoophm) [inline src/app/client/src/app/app.module.ts]: shared-feature

File: src/app/client/src/app/modules/learn/learn-routing.module.ts [hunk 3]
```diff
@@ -10,22 +10,33 @@ import { ResourceService } from '@sunbird/shared';
...
 const telemetryEnv = 'course';
 const objectType = 'course';
 const routes: Routes = [
   {
     path: '', component: LearnPageComponent,
     data: {
       breadcrumbs: [{ label: 'Home', url: '/home' }, { label: 'Courses', url: '' }],
-      telemetry: { env: telemetryEnv, pageid: 'learn', type: 'view' }
+      telemetry: { env: telemetryEnv, pageid: 'learn', type: 'view' },
+      url: 'learn'
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/learn-routing.module.ts]: baseUrl

File: src/app/client/src/app/modules/learn/learn-routing.module.ts [hunk 4]
```diff
@@ -10,22 +10,33 @@ import { ResourceService } from '@sunbird/shared';
...
-  },
+    },
   {
     path: 'redirect', component: RedirectComponent,
     data: {
       telemetry: { env: telemetryEnv, pageid: 'learn-redirect', type: 'view' }
     }
   },
+  {
+    path: 'view-All/:sectionName/:pageNumber', component: ViewAllComponent,
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/learn-routing.module.ts]: section

File: src/app/client/src/app/modules/learn/learn-routing.module.ts [hunk 5]
```diff
@@ -10,22 +10,33 @@ import { ResourceService } from '@sunbird/shared';
...
+    data: {
+      breadcrumbs: [{ label: 'Home', url: '/home' }, { label: 'Courses', url: '/learn' }],
+      telemetry: {
+        env: 'viewAll', pageid: 'viewAll', type: 'view', subtype: 'paginate'
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/learn-routing.module.ts]: env should be learn

File: src/app/client/src/app/modules/learn/learn.module.ts [hunk 6]
```diff
@@ -29,7 +29,8 @@ import { DashboardModule } from '@sunbird/dashboard';
     ReactiveFormsModule,
     NotesModule,
     TelemetryModule,
-    NgInviewModule
+    NgInviewModule,
+    SharedFeaturedModule
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/learn.module.ts]: include this in app.module

File: src/app/client/src/app/modules/shared-featured/shared-featured-routing.module.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import { NgModule } from '@angular/core';
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared-featured/shared-featured-routing.module.ts]: routing module is not need for this

File: src/app/client/src/app/modules/shared/components/page-section/page-section.component.html [hunk 8]
```diff
@@ -1,6 +1,7 @@
 <div class="ui link popular-section" *ngIf="section.contents && section.contents.length > 0">
     <h4 class="ui header card-section-header sectionHeading">{{section.name}}
         <span class="ui red circular inline label">{{section.contents.length}}</span>
+        <button *ngIf="section.contents.length >= 10 && section.name != My Courses " type="button" (click)="redirect()">more</button>
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared/components/page-section/page-section.component.html]: dont redirect from here

File: src/app/client/src/app/modules/shared/components/page-section/page-section.component.ts [hunk 9]
```diff
@@ -159,4 +160,10 @@ export class PageSectionComponent implements OnInit {
       this.visits.emit(visits);
     }
   }
+  redirect() {
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared/components/page-section/page-section.component.ts]: emit event from pagesection and subscribe to it in the parent. on getting event redirect.

File: src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts [hunk 10]
```diff
@@ -86,6 +86,7 @@ export class LearnPageComponent implements OnInit, OnDestroy {
   content: any;
   public unsubscribe = new Subject<void>();
   courseDataSubscription: Subscription;
+  sectionUrl: string;
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts]: why do we need this as a property

File: src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts [hunk 11]
```diff
@@ -308,6 +309,12 @@ export class LearnPageComponent implements OnInit, OnDestroy {
     }
     this.playerService.playContent(event.data.metaData);
   }
+  viewAll(event) {
+    this.activatedRoute.data.subscribe(data => {
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts]: dont take this from router.data, you can add this to config or hardcode the value here, as we know we are in learn page

File: src/app/client/src/app/modules/shared/components/page-section/page-section.component.html [hunk 12]
```diff
@@ -1,6 +1,7 @@
 <div class="ui link popular-section" *ngIf="section.contents && section.contents.length > 0">
     <h4 class="ui header card-section-header sectionHeading">{{section.name}}
         <span class="ui red circular inline label">{{section.contents.length}}</span>
+        <button *ngIf="section.contents.length >= 10 && section.name != My Courses " type="button" (click)="redirect(section)">more</button>
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared/components/page-section/page-section.component.html]: navigateToViewAll

File: src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts [hunk 13]
```diff
@@ -308,6 +308,10 @@ export class LearnPageComponent implements OnInit, OnDestroy {
     }
     this.playerService.playContent(event.data.metaData);
   }
+  viewAll(event) {
+    const sectionUrl = 'learn/view-All/' + event.name;
```
Comment (anoophm) [inline src/app/client/src/app/modules/learn/components/learn-page/learn-page.component.ts]: view-all

File: src/app/client/src/app/modules/shared/components/page-section/page-section.component.html [hunk 14]
```diff
@@ -1,6 +1,7 @@
 <div class="ui link popular-section" *ngIf="section.contents && section.contents.length > 0">
     <h4 class="ui header card-section-header sectionHeading">{{section.name}}
         <span class="ui red circular inline label">{{section.contents.length}}</span>
+        <button *ngIf="section.contents.length >= 10 && section.name != My Courses " type="button" (click)="navigateToViewAll(section)">more</button>
```
Comment (anoophm) [inline src/app/client/src/app/modules/shared/components/page-section/page-section.component.html]: they want view all for my course as well right

Comment (ArchanaKumari01) [reply src/app/client/src/app/modules/shared/components/page-section/page-section.component.html]: No, they only want it for other courses which have the length of courses more than or equal to 10.

**Pull request merged**